source/ui/CLS/CLSSIS3820ScalerChannelViewWithDarkCurrent.cpp

```diff
@@ -51,91 +50,76 @@ CLSSIS3820ScalerChannelViewWithDarkCurrent::CLSSIS3820ScalerChannelViewWithDarkC
+void CLSSIS3820ScalerChannelViewWithDarkCurrent::setDarkCurrentViewMode(DarkCurrentViewMode newMode)
 {
-	return darkCurrentViewMode_;
-}
+	if (channel_ && channel_->detector() && channel_->detector()->canDoDarkCurrentCorrection()) {
 
+		if (darkCurrentViewMode_ != newMode) {
+			darkCurrentViewMode_ = newMode;
 
+			if (darkCurrentViewMode_ == Hide) {
+				channelLayout_->removeItem(content_);
 
-void CLSSIS3820ScalerChannelViewWithDarkCurrent::setDarkCurrentViewMode(DarkCurrentViewMode newMode)
-{
-	darkCurrentViewMode_ = newMode;
+			} else if (darkCurrentViewMode_ == Show) {
+				channelLayout_->addLayout(content_);
 
-	if (darkCurrentViewMode_ == Hide) {
-		channelLayout_->removeItem(content_);
-		emit darkCurrentViewModeChanged(darkCurrentViewMode_);
+			} else {
+				darkCurrentViewMode_ = Hide;
+			}
 
-	} else if (darkCurrentViewMode_ == Show) {
-		channelLayout_->addLayout(content_);
-		emit darkCurrentViewModeChanged(darkCurrentViewMode_);
-
-	} else {
-		setDarkCurrentViewMode(Hide);
+			emit darkCurrentViewModeChanged(darkCurrentViewMode_);
+		}
 	}
 }
 
-
-
-void CLSSIS3820ScalerChannelViewWithDarkCurrent::onNewDarkCurrentMeasurementValue(double newValue)
+void CLSSIS3820ScalerChannelViewWithDarkCurrent::setDarkCurrentValue(double newValue)
 {
-	setDarkCurrentValueLabel(newValue);
+	darkCurrentValueLabel_->setText(QString("Dark current value : %1").arg((int)newValue));
 }
 
+void CLSSIS3820ScalerChannelViewWithDarkCurrent::setDarkCurrentCorrected(double newValue)
+{
+	darkCurrentCorrectedLabel_->setText(QString("Corrected measurement : %1").arg((int)newValue));
+}
 
-
-void CLSSIS3820ScalerChannelViewWithDarkCurrent::onNewDarkCurrentMeasurementState(bool measurementUpToDate)
+void CLSSIS3820ScalerChannelViewWithDarkCurrent::setDarkCurrentState(bool valid)
 {
-	if (measurementUpToDate)
+	if (valid)
 		darkCurrentValueLabel_->setStyleSheet("color: blue;");
 
 	else
 		darkCurrentValueLabel_->setStyleSheet("color: red;");
 }
 
-
-
-void CLSSIS3820ScalerChannelViewWithDarkCurrent::onDetectorNewValuesAvailable()
+void CLSSIS3820ScalerChannelViewWithDarkCurrent::onDetectorAcquisitionStateChanged(AMDetector::AcqusitionState newState)
 {
-	// we arrive at a dark current corrected value by dividing the number of counts aquired by the measurement dwell time, and subtract the dark current.
-	double *newValue = 0;
-	channel_->detector()->data(newValue);
-
-	double dwellTime = channel_->detector()->acquisitionTime();
-
-	// the latest value of the detector's dark current. this is normalized by its acquisition time already (see AMDetector::setAsDarkCurrentMeasurementValue()).
-	double darkCurrent = channel_->detector()->darkCurrentMeasurementValue();
-
-	double correctedMeasurement_ = *newValue / dwellTime - darkCurrent;
-
-	qDebug() << "CLSSIS3820ScalerChannelView has new DC corrected value : " << correctedMeasurement_;
-
-	setCorrectedMeasurementLabel(correctedMeasurement_);
+	switch (newState) {
+	case AMDetector::Succeeded:
```

Comment:
Do we not need to check the other states?  I'm thinking Failed and Cancelled in particular.